src/zenml/utils/code_utils.py

```diff
@@ -168,55 +169,69 @@ def write_archive(
             )
 
 
-def upload_code_if_necessary(code_archive: CodeArchive) -> str:
-    """Upload code to the artifact store if necessary.
-
-    This function computes a hash of the code to be uploaded, and if an archive
-    with the same hash already exists it will not re-upload but instead return
-    the path to the existing archive.
+def zip_and_hash_code(code_archive: CodeArchive) -> Tuple[str, str]:
+    """Zip the code archive and compute its hash.
 
     Args:
-        code_archive: The code archive to upload.
+        code_archive: The code archive to zip and hash.
 
     Returns:
-        The path where to archived code is uploaded.
+        A tuple containing the temporary file path and the computed hash.
```

Comment:
The second item of this tuple is not the computed hash, but instead a filename. I think returning two filenames is not intuitive at all what they're supposed to be.

As far as I can see, this function is used below and in the lightning orchestrator, which actually doesn't even use the hash.

I would suggest we just use 
```python
with tempfile.NamedTemporaryFile(
        mode="w+b", delete=False, suffix=".tar.gz"
    ) as f:
        code_archive.write_archive(f)
```
in the lightning orchestrator (and also in the function below), I don't think that's worth having in a separate helper function.

I still like your idea of extracting the hash computation though, maybe we can make this `compute_file_hash(file) -> str`?

Reply:
I think it's just doc string error later on i was also using the filename as second output

Reply:
Still I think it would be nicer to have a helper function that computes just the hash of a file instead of having both the archive writing and hash computation combined in one function.
If we had a util function to compute just the file hash, we have other places in the code where we could reuse it.

If you still think that we should not duplicate this code

```python
with tempfile.NamedTemporaryFile(
        mode="w+b", delete=False, suffix=".tar.gz"
    ) as f:
        code_archive.write_archive(f)
```
maybe we create a second helper function which is called `write_archive_to_temporary_file(...) -> str` which does this and returns the filename?